# 1. Fix linking on Windows.
# 2. Use assembly optimizations on Linux-aarch64 (https://boringssl-review.googlesource.com/c/boringssl/+/47084).
--- a/BUILD
+++ b/BUILD
@@ -18,6 +18,7 @@ load(
     "crypto_headers",
     "crypto_internal_headers",
     "crypto_sources",
+    "crypto_sources_linux_aarch64",
     "crypto_sources_linux_ppc64le",
     "crypto_sources_linux_x86_64",
     "crypto_sources_mac_x86_64",
@@ -34,6 +35,11 @@ licenses(["notice"])
 exports_files(["LICENSE"])

 config_setting(
+    name = "linux_aarch64",
+    values = {"cpu": "aarch64"},
+)
+
+config_setting(
     name = "linux_x86_64",
     values = {"cpu": "k8"},
 )
@@ -103,17 +109,20 @@ posix_copts = [
 ]

 boringssl_copts = select({
+    ":linux_aarch64": posix_copts,
     ":linux_ppc64le": posix_copts,
     ":linux_x86_64": posix_copts,
     ":mac_x86_64": posix_copts,
     ":windows_x86_64": [
         "-DWIN32_LEAN_AND_MEAN",
         "-DOPENSSL_NO_ASM",
+        "-DBORINGSSL_IMPLEMENTATION",
     ],
     "//conditions:default": ["-DOPENSSL_NO_ASM"],
 })

 crypto_sources_asm = select({
+    ":linux_aarch64": crypto_sources_linux_aarch64,
     ":linux_ppc64le": crypto_sources_linux_ppc64le,
     ":linux_x86_64": crypto_sources_linux_x86_64,
     ":mac_x86_64": crypto_sources_mac_x86_64,
@@ -129,6 +138,7 @@ posix_copts_c11 = [
 ]

 boringssl_copts_c11 = boringssl_copts + select({
+    ":linux_aarch64": posix_copts_c11,
     ":linux_ppc64le": posix_copts_c11,
     ":linux_x86_64": posix_copts_c11,
     ":mac_x86_64": posix_copts_c11,
@@ -142,6 +152,7 @@ posix_copts_cxx = [
 ]

 boringssl_copts_cxx = boringssl_copts + select({
+    ":linux_aarch64": posix_copts_cxx,
     ":linux_ppc64le": posix_copts_cxx,
     ":linux_x86_64": posix_copts_cxx,
     ":mac_x86_64": posix_copts_cxx,
@@ -166,6 +177,7 @@ cc_library(
         ":windows_x86_64": ["-defaultlib:advapi32.lib"],
         "//conditions:default": ["-lpthread"],
     }),
+    linkstatic = True,
     visibility = ["//visibility:public"],
 )

@@ -175,6 +187,7 @@ cc_library(
     hdrs = ssl_headers,
     copts = boringssl_copts_cxx,
     includes = ["src/include"],
+    linkstatic = True,
     visibility = ["//visibility:public"],
     deps = [
         ":crypto",
